### ISSUE TITLE
Suggestion: Requirements on node version should be a breaking change/be a major version bump

### ISSUE THREAD

**Issue by vandercloak** [semantic-release/semantic-release#2146]
Was blind sided today when all of our CI releases stopped working with node version <14.17. The change made in the latest release that added this requirement feels like it should have been a MAJOR/breaking change and I wanted to suggest that for the future. 

![image](https://user-images.githubusercontent.com/7164984/134239138-ddb2c550-fd5b-456d-bf98-a8c38daf1613.png)

**Comment by travi** [comment]
all of the node requirement increases have been released as breaking changes, such as https://github.com/semantic-release/semantic-release/releases/tag/v18.0.0. could you provide more detail around where you are seeing this not handled as expected?

**Comment by vandercloak** [comment]
@travi 🤦‍♂️ That was stupid of me. Definitely jumped the gun. I saw the commit and thought the release was based off of that commit message (shown in image). 

Do you know why I would be hitting this error `[semantic-release]: node version >=14.17 is required. Found v12.22.6.`  when I am locked to semantic release version 17.4.2?

From my package.json (the lock file is also updated):
```
"semantic-release": "17.4.2",
"semantic-release-slack-bot": "2.1.0",
```

**Comment by travi** [comment]
how are you executing semantic-release? are you using an approach that would use an instance that is different than the one you define in your package.json?

**Comment by vandercloak** [comment]
Yeah, just confirmed it. I reverted back to `"@semantic-release/git": "^9.0.0"` and then got the node version error. Then reimplemented `"@semantic-release/git": "9.0.0"` and the build worked. I am digging through the @semantic-release/git latest release tag dif and not sure yet what caused it.

**Comment by travi** [comment]
the git plugin should have no impact to the error reported. the node version check and message [are from the core semantic-release package](https://github.com/semantic-release/semantic-release/blob/971a5e0d16f1a32e117e9ce382a1618c8256d0d9/bin/semantic-release.js#L15-L22).

what do the following commands report for you?
* `npm ls semantic-release`
* `npm ls @semantic-release/git`

also 

> how are you executing semantic-release? are you using an approach that would use an instance that is different than the one you define in your package.json?

**Comment by triwats** [comment]
Seeing oddities with this recent release around dependencies. We use a CI/CD template that was working until 4 days ago in-line with the 18.0.0 release. Can't quite put my finger on why this isn't working properly as it seemingly can find the intended packages, but we end up in issues with dependancies. Cleared the cache to no avail, we should be pinning anyway so gone and done so. 

We're using Docker in CI/CD and running fresh installs:

Node Version: 16.5

This is the script we were using to do our releases:

```
before_script:
  - npm install semantic-release
  - npm install @semantic-release/commit-analyzer
  - npm install @semantic-release/release-notes-generator
  - npm install @semantic-release/gitlab
  - npm install @semantic-release/changelog
  - npm install @semantic-release/git
```

We had to pin the following packages:

```
1. @semantic-release/commit-analyzer
2.@semantic-release/release-notes-generator
3. @semantic-release/gitlab
4. @semantic-release/git
```

All of the above packages resulted in the following error:

```
$ npm install @semantic-release/git
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: newstore-import-poller@0.0.1
npm ERR! Found: semantic-release@17.4.7
npm ERR! node_modules/semantic-release
npm ERR!   semantic-release@"^17.4.7" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer semantic-release@">=18.0.0" from @semantic-release/git@10.0.0
npm ERR! node_modules/@semantic-release/git
npm ERR!   @semantic-release/git@"*" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /root/.npm/eresolve-report.txt for a full report.
npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2021-09-21T22_15_14_622Z-debug.log
```

Hope this helps rather than confusing anything! Let me know if you need a hand with anything, but fairly new with `npm`!

**Comment by travi** [comment]
@triwats your issue is different enough from this thread that i have moved it to a separate issue

**Comment by vandercloak** [comment]
![image](https://user-images.githubusercontent.com/7164984/134531473-4d323f5f-ef25-4150-8438-09b3789c741d.png)
![image](https://user-images.githubusercontent.com/7164984/134531563-1312fad6-d322-4800-8954-31054399f202.png)

> how are you executing semantic-release? are you using an approach that would use an instance that is different than the one you define in your package.json?

We are executing semantic release inside a github action. 

```
jobs:
  release:
    name: Publish 🚀
    runs-on: ubuntu-latest
    strategy:
      matrix:
        node-version: [12.x]

    steps:
      - uses: actions/checkout@v2
      
      # ...

      - name: 🟢 Node
        uses: actions/setup-node@v1
        with:
          node-version: ${{ matrix.node-version }}
          registry-url: "https://registry.npmjs.org"

      - name: 📚 Cache
        uses: actions/cache@v2
        env:
          cache-name: cache-node-modules
        with:
          # npm cache files are stored in `~/.npm` on Linux/macOS
          path: ~/.npm
          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
          restore-keys: |
            ${{ runner.os }}-build-${{ env.cache-name }}-
            ${{ runner.os }}-build-
            ${{ runner.os }}-

      - name: 🛠 Install
        run: npm install

      - name: ✅ Test
        working-directory: ${{ github.workspace }}
        run: npm run test

      - name: 🏗 Build
        run: npm run build

      - name: 🚀 Publish Rainbow
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        run: cd dist && SEMANTIC_RELEASE_PACKAGE=🌈 npx semantic-release
```

**Comment by travi** [comment]
by default, `npx` should use the locally installed packages before pulling a remote version. your change to `dist` could introduce different behavior, but i would normally expect it to still find the packages from the parent directory. what is the reason that you are changing directories for release? does the `dist` directory exist because of some build step that is separate from the release process managed by semantic-release? our recommendation is normally to let semantic-release coordinate the entire release process, including any build steps.

if you aren't already, it may be useful to run `semantic-release` using the `--debug` flag to get more output to confirm things like versions that are used during the release execution.

also, i'm all for tracking down the reason that this is giving you trouble, but i'm curious: is there something preventing you from upgrading the node version used for the release steps? it looks like you are using a matrix for node versions, so we normally recommend separating the release from the matrix job. if separated, you can set the node version specifically for that job and we recommend using `lts/*` so that it adjusts over time as new lts versions are released

**Comment by vandercloak** [comment]
Looks like the cd into dist was the issue. Double checked that I was looking at the right github action this time haha. 

![image](https://user-images.githubusercontent.com/7164984/134539110-e7e3252e-bcb9-47d4-8f25-19cb0403fbb9.png)

---
> what is the reason that you are changing directories for release

We had a old build process that we worked semantic-release into vs starting from scratch with semantic-release. I could see us moving toward fully utilizing semantic-release for the whole process in the future. 

> if you aren't already, it may be useful to run semantic-release using the --debug flag to get more output to confirm things like versions that are used during the release execution.

This is great. I will remember to use next time we run into issue.

> also, i'm all for tracking down the reason that this is giving you trouble, but i'm curious: is there something preventing you from upgrading the node version used for the release steps?

We have some old packages that depend on older versions... kinda a snowball of issues there but we are working to get up to node 14+. 

Thanks again for the help and suggestions!

**Comment by travi** [comment]
> We have some old packages that depend on older versions... kinda a snowball of issues there but we are working to get up to node 14+.

is the node version limiting for those packages as part of the release? my point is that we recommend separating the release step from your verification steps or other reasons where your packages have node version requirements that are different than semantic-release. actions enables this separation quite well, so there is usually very little reason to perform release tasks on a different node version than the latest lts.

for example, you can see how we run a matrix job separately from a job on a specific version for the verification of this repo: https://github.com/semantic-release/semantic-release/blob/master/.github/workflows/test.yml. we separate the release to a completely separate workflow for this project, which enables us to use a specific version there as well. (we have not updated to our recommendation of using `lts/*` as that version within this project, but imagine the 16 being `lts/*` in those workflows).

as another example, i often verify and release from the same workflow in some of my other projects, but again separate the matrix for verification and then define a compatible version for release, like https://github.com/form8ion/javascript-core/blob/master/.github/workflows/node-ci.yml

**Comment by vandercloak** [comment]
Oh I see what you mean. If that's the case then there shouldn't be anything holding us back. I will try and update
